src/main/java/com/elixr/poc/service/PurchaseService.java

```diff
@@ -0,0 +1,44 @@
+package com.elixr.poc.service;
+import com.elixr.poc.data.Purchase;
+import com.elixr.poc.repository.PurchaseRepository;
+import com.elixr.poc.rest.request.PurchaseRequest;
+import com.elixr.poc.rest.response.PurchaseResponse;
+import org.springframework.stereotype.Service;
+
+import java.util.UUID;
+
+/**
+ * service class for editing response and save or update new purchase to database
+ */
+
+@Service
+public class PurchaseService {
+    private final PurchaseRepository purchaseRepository;
+
+    public PurchaseService(PurchaseRepository purchaseRepository) {
+        this.purchaseRepository = purchaseRepository;
+    }
+
+    private static Purchase createPurchaseObjectFromRequest(PurchaseRequest purchaseRequest) {
+        return Purchase.builder().userName(purchaseRequest.getUserName()).product(purchaseRequest.getProduct()).amount(purchaseRequest.getAmount()).date(purchaseRequest.getDate()).build();
+    }
+
+    public PurchaseResponse createPurchase(PurchaseRequest purchaseRequestObject) {
+        Purchase purchaseObject = createPurchaseObjectFromRequest(purchaseRequestObject);
+        saveRepository(purchaseObject);
+        return PurchaseResponse.purchaseBuilder().success(true).id(purchaseObject.getId()).userName(purchaseObject.getUserName()).product(purchaseObject.getProduct()).amount(purchaseObject.getAmount()).date(purchaseObject.getDate()).build();
+    }
+
+    /**
+     * Calling purchase repository to store the valid user information to the database
+     * @param purchase
+     * @return
+     */
+    private Purchase saveRepository(Purchase purchase) {
```

Comment:
saveRepository name suggests you are saving respository object which is wrong.
either rename it as savePurchaseInDB or savePurchase

Reply:
Updated